rslib/src/search/sqlwriter.rs

```diff
@@ -184,16 +185,69 @@ impl SqlWriter<'_> {
         Ok(())
     }
 
-    fn write_unqualified(&mut self, text: &str) {
+    /// Return excluded fields for notetypes that have some but not all fields
+    /// excluded
+    #[allow(clippy::type_complexity)]
+    fn excluded_fields_indices_by_notetype(
+        &mut self,
+    ) -> Result<Option<Vec<(NotetypeId, u32, HashSet<u32>)>>> {
+        let notetypes = self.col.get_all_notetypes()?;
+        let mut any_excluded = false;
+        let mut field_map = vec![];
+        for nt in notetypes.values() {
```

Comment:
In Rust it's preferred to map (or flatmap) the iterator then collect it, instead of creating a separate vec[] and looping over an iterator.